File: model/daily.go
package model

import "go.mongodb.org/mongo-driver/bson/primitive"

type Daily struct {
	ID         primitive.ObjectID   `json:"id" bson:"_id"`
	Text       string               `json:"text" bson:"text"`
	Author     primitive.ObjectID   `json:"author" bson:"author"`
	Keywords   []string             `json:"keywords" bson:"keywords"`
	Emotions   Emotion              `json:"emotions" bson:"emotions"`
	Image      string               `json:"image" bson:"image"`
	Favourites int                  `json:"favourites" bson:"favourites"`
	CreatedAt  primitive.DateTime   `json:"createdAt" bson:"createdAt"`
	Viewers    []primitive.ObjectID `json:"viewers" bson:"viewers"`
	IsShared   bool                 `json:"isShared" bson:"isShared"`
}

type Emotion struct {
	anger int
	happy int
	sad   int
	shock int
}

File: daily-ui/libs/types/daily.ts
export type CreateDailyRequest = {
   image?: string,
   text: string,
   isShared: boolean,
   prompt: string
};

export type StatisticsResponse = {
   date: string[],
   likes: number,
   views: number,
   dailiesWritten: number,
   mood: string,
   streak: number,
   topics: string[]
};

export type ExploreResponse = {
   id: string,
   text: string,
   image: string,
};


export type DailyResponse = {
   id: string,
   text: string,
   author: string,
   topics: string[],
   keywords: string[],
   emotions: {
      anger: number,
      happiness: number,
      sadness: number,
      shock: number
   },
   image: string,
   favourites: number,
   createdAt: string,
   viewers: [],
   isshared: boolean
};

export type EditDailyImageRequest = {
   id: string,
   image: string
};

export type ReportDailyRequest = {
   content: string,
   dailyId: string,
   title: string
};
